pkg/repro/repro.go

```diff
@@ -421,13 +435,18 @@ func (ctx *context) simplifyProg(res *Result) (*Result, error) {
 		ctx.stats.SimplifyProgTime = time.Since(start)
 	}()
 
-	opts := res.Opts
-	opts.Collide = false
-	crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
-	if err != nil {
-		return res, err
+	if res.ops.Collide {
+		opts := res.Opts
+		opts.Collide = false
+		crashed, err := ctx.testProg(res.Prog, res.Duration, opts)
+		if err != nil {
+			return res, err
+		}
+		if crashed {
+			res.Opts = opts
+		}
 	}
-	if crashed {
+	if !res.ops.Collide && res.ops.Threaded {
 		res.Opts = opts
```

Comment:
First we try to remove `collide`. If that succeeds, we try to remove `threaded`, otherwise we keep threaded. At least that's what it seemed to do before, I just tried to make the code more alike to the rest of the function here.